src/librustc_trans/trans/common.rs

```diff
@@ -213,8 +213,27 @@ pub fn type_needs_unwind_cleanup<'a, 'tcx>(ccx: &CrateContext<'a, 'tcx>, ty: Ty<
     }
 }
 
+
+/// Issue #22815: call the `FunctionContext::type_needs_drop` method
+/// instead, if possible; it is strictly more precise than this
+/// function.
 pub fn type_needs_drop<'tcx>(cx: &ty::ctxt<'tcx>, ty: Ty<'tcx>) -> bool {
```

Comment:
I guess this was only implicit in the PR: the two cases where I left calls to `common::type_needs_drop` were because there is not a `FunctionContext` readily available.  I spent a while tracing back through the call chain to see if I could refactor the code to pass down a `Block` (with its attached `FunctionContext`) instead of passing down a `CrateContext` in each of the two cases, but in both of them as I worked my way backwards it seemed not feasible. (In particular, in one case the call chain goes back to translating const expressions, where I assume it is by design that there is no `FunctionContext` available.)